neo/rawio/plexon2rawio/plexon2rawio.py

```diff
@@ -0,0 +1,498 @@
+        assert stream_id == stream_characteristic.id
+        return stream_characteristic.n_samples
+
+    def _get_signal_t_start(self, block_index, seg_index, stream_index):
+        # This give the t_start of signals.
+        # this must return an float scale in second
+
+        # TODO: Does the fragment_timestamp[0] need to be added here as for digital signals?
+        return self._segment_t_start(block_index, seg_index)
+
+    def _get_analogsignal_chunk(self, block_index, seg_index, i_start, i_stop,
+                                stream_index, channel_indexes):
+        # this must return a signal chunk in a signal stream
+        # limited with i_start/i_stop (can be None)
+        # channel_indexes can be None (=all channel in the stream) or a list or numpy.array
+        # This must return a numpy array 2D (even with one channel).
+        # This must return the original dtype. No conversion here.
+        # This must as fast as possible.
+        # To speed up this call all preparatory calculations should be implemented
+        # in _parse_header().
+
+        stream_id = self.header['signal_streams'][stream_index]['id']
+        mask = self.header['signal_channels']['stream_id'] == stream_id
+        stream_channels = self.header['signal_channels'][mask]
+
+        n_channels = len(stream_channels)
+        n_samples = self.get_signal_size(block_index, seg_index, stream_index)
+
+        if i_start is None:
+            i_start = 0
+        if i_stop is None:
+            i_stop = n_samples
+
+        if i_start < 0 or i_stop > n_samples:
+            raise IndexError(f"Indexes ({i_start}, {i_stop}) out of range for signal with {n_samples} samples")
+
+        # converting channel_indexes to array representation
+        if channel_indexes is None:
+            channel_indexes = np.arange(len(stream_channels), dtype='int')
+        elif isinstance(channel_indexes, slice):
+            channel_indexes = np.arange(len(stream_channels), dtype='int')[channel_indexes]
+        else:
+            channel_indexes = np.asarray(channel_indexes)
+
+        # channel index sanity check
+        if any(channel_indexes < 0) or any(channel_indexes >= n_channels):
+            raise IndexError(f'Channel index out of range {channel_indexes} for stream with {n_channels} channels')
+
+        nb_chan = len(channel_indexes)
+
+        raw_signals = np.empty((i_stop - i_start, nb_chan), dtype='int16')
+        for i, channel_idx in enumerate(channel_indexes):
+            channel_name = stream_channels['name'][channel_idx]
+
+            # use previously loaded channel data if possible
+            if channel_name in self._analogsignal_cache:
+                values = self._analogsignal_cache[channel_name]
+            else:
+                res = self.pl2reader.pl2_get_analog_channel_data_by_name(channel_name)
+                fragment_timestamps, fragment_counts, values = res
+                self._analogsignal_cache[channel_name] = values
```

Comment:
I guess `pl2_get_analog_channel_data_by_name` is very slow...

Reply:
I did some test runs of this on a multi-GB pl2 file and it seems that the `_by_name` method is even a tiny bit faster than accessing the channel by its id. I don't know what exactly is happening in the back as the dll is closed source, but I would stick to the `pl2_get_analog_channel_dat_by_name` method here for now.